libs/orders/src/lib/components/order-data-provider/order-data-provider.ts

```diff
@@ -103,7 +103,7 @@ export const mapOrderUpdateToOrder = (
       id: marketId,
     },
     __typename: 'Order',
-  };
+  } as OrderFieldsFragment;
```

Comment:
Something seems wrong with the types here. The return type is already defined so why do we need to type cast it again later?

Reply:
Its a little ugly but this will solve it

```
icebergOrder: order.icebergOrder
      ? {
          __typename: 'IcebergOrder',
          ...order.icebergOrder,
        }
      : undefined,
```

Reply:
Part of the problem is because our type generation sets the __typename as undefined. We should be able to edit our type generation config so we dont get undefined for typenames

Reply:
awesome, I'm removing the cast. wasn't a fan either